packages/react-wildcat-handoff/src/utils/serverRender.js

```diff
@@ -142,3 +144,7 @@ function getHtmlNotFoundTemplate(serverSettings) {
         status: 404
     };
 }
+
+function has404StatusRouterProperty(components) {
+    return components.some(component => component.routerProps && component.routerProps.status === 404);
```

Comment:
components are ordered in render order. Since each component can define a `routerProps` static, you should pluck the last component that has a static defined and return that.